BacterialWrapperScript/BacterialWrapperScript.py

```diff
@@ -0,0 +1,295 @@
+#                 for line in fh:
+#                     if not line.startswith(b"#"):
+#                         out_f.write(line)
+
+def stitch_all_outputs(ofw: OutputFileWriter, output_files: list[tuple[int, dict[str, Path]]], 
+                       threads: int | None = None) -> None:
+    
+    fq1_list = []
+    fq2_list = []
+    bam_list = []
+
+    for (thread_idx, file_dict) in output_files:
+        if file_dict["fq1"]:
+            fq1_list.append(file_dict["fq1"])
+        if file_dict["fq2"]:
+            fq2_list.append(file_dict["fq2"])
+        if file_dict["bam"]:
+            bam_list.append(file_dict["bam"])
+    
+    concat_fq(fq1_list, ofw.files_to_write[ofw.fq1])
+    concat_fq(fq2_list, ofw.files_to_write[ofw.fq2])
+    merge_bam(bam_list, ofw, threads)
+
+# def stitch_all_outputs(options: Options, thread_options: list[Options]) -> None:
+#     fq1_list = []
+#     fq2_list = []
+#     vcf_list = []
+#     bam_list = []
+
+#     for local_ops in thread_options:
+#         if local_ops.fq1:
+#             fq1_list.append(local_ops.fq1)
+#         if local_ops.fq2:
+#             fq2_list.append(local_ops.fq2)
+#         if local_ops.vcf:
+#             vcf_list.append(local_ops.vcf)
+#         if local_ops.bam:
+#             bam_list.append(local_ops.bam)
+    
+#     concat_fq(fq1_list, options.fq1)
+#     concat_fq(fq2_list, options.fq2)
+#     merge_bam(bam_list, options.bam)
+#     merge_vcf(vcf_list, options.vcf)
+
+
+# orig_salmonella_input = Path("Outputs/Salmonella/salmonella_config_test.fastq")
+# new_salmonella_input = Path("Outputs/Salmonella/new_Salmonella_config_test.fastq")
+# salmonella_inputs = [orig_salmonella_input, new_salmonella_input]
+# dest = Path("Outputs/Salmonella/salmonella_stitched.fastq")
+# concat_fq(salmonella_inputs, dest)
+
+# orig_salmonella_input = Path("Outputs/Salmonella/salmonella_config_test_golden.bam")
+# new_salmonella_input = Path("Outputs/Salmonella/new_Salmonella_config_test_golden.bam")
+# salmonella_inputs = [orig_salmonella_input, new_salmonella_input]
+# dest = Path("Outputs/Salmonella/salmonella_stitched_golden.bam")
+# merge_bam(salmonella_inputs, dest)
+
+# orig_salmonella_input = Path("Outputs/Salmonella/salmonella_config_test_golden.vcf.gz")
+# new_salmonella_input = Path("Outputs/Salmonella/new_Salmonella_config_test_golden.vcf.gz")
+# salmonella_inputs = [orig_salmonella_input, new_salmonella_input]
+# dest = Path("Outputs/Salmonella/salmonella_stitched_golden.vcf.gz")
+# merge_vcf(salmonella_inputs, dest)
+
+# output_path = Path("Outputs/Salmonella/all_stitched/final")
+# config_file = salmonella_config_file
+# reference = salmonella_file
+# fq1 = Path("Outputs/Salmonella/salmonella_config_test.fastq")
+# fq2 = Path("")
+# vcf = Path("Outputs/Salmonella/salmonella_config_test_golden.vcf.gz")
+# bam = Path("Outputs/Salmonella/salmonella_config_test_golden.bam")
+
+
+# options = Options(reference)
+# bam_header = {}
+
+# ofw = OutputFileWriter(options, bam_header)
```

Comment:
You can do it this way, if you get all the parts to work, or just directly import Bio.bgzf and open the files with bgzfReader and bgzfWriter. Whichever way is easiest for now is perfect.